pandas/core/indexes/category.py

```diff
@@ -341,9 +341,26 @@ def __array__(self, dtype=None):
 
     @Appender(_index_shared_docs['astype'])
     def astype(self, dtype, copy=True):
+        dtype = pandas_dtype(dtype)
         if is_interval_dtype(dtype):
             from pandas import IntervalIndex
             return IntervalIndex.from_intervals(np.array(self))
+        elif is_categorical_dtype(dtype):
+            # want to maintain existing categories/ordered if they are None
+            if dtype.categories is None:
+                new_categories = self.categories
+            else:
+                new_categories = dtype.categories
+            if dtype.ordered is None:
+                new_ordered = self.ordered
+            else:
+                new_ordered = dtype.ordered
+            dtype = CategoricalDtype(new_categories, new_ordered)
+
+            # fastpath if dtypes are equal
+            if dtype == self.dtype:
```

Comment:
Note that `dtype.equals(self.dtype)` raises `AttributeError: 'CategoricalDtype' object has no attribute 'equals'`.

The only other thing I found for dtype comparison is `is_dtype_equal`, but under the hood that basically just does `==` within a `try`/`except` to catch if the dtypes aren't comparable.  We're within `elif is_categorical_dtype(dtype)` here though, so `is_dtype_equal` seem superfluous, but could still use it if preferred.